stages/org.osbuild.oscap.remediation

```diff
@@ -128,15 +151,131 @@ def main(tree, options):
     if tailoring is not None:
         cmd.extend(["--tailoring-file", tailoring])
 
+    if arf_results is not None:
+        # run with command in chroot so full path ok
+        cmd.extend(["--results-arf", f"{data_dir}/{arf_results}"])
+
+    cmd.extend(["--results", f"{data_dir}/{xccdf_results}"])
+
     cmd.append(datastream)
 
-    res = subprocess.run(cmd, encoding="utf8", stdout=sys.stderr, check=False)
+    print("OpenSCAP (evaluate): ", repr(cmd))
+    # we need to
+    # 1) inform the scanner that the system is offline (OSCAP_PROBE_ROOT="")
+    # the variable will make OVAL probes to be less affected by the absence
+    # of dynamic things like /proc /dev etc and also will define the path prefix
+    # (we keep it empty since our process is already chrooted into the image filesystem)
+    # 2) provide important environment variables in OSCAP_CONTAINER_VARS as
+    # the probe does not have access to /proc
+    res = subprocess.run(cmd, encoding="utf8", stdout=sys.stderr, check=False,
+                         env=dict(os.environ,
+                                  OSCAP_PROBE_ROOT="",
+                                  OSCAP_CONTAINER_VARS="container=bwrap-osbuild"))
 
     # oscap return values are:
     # 0 → success
-    # 2 → -- no error, but some checks/remediation failed
+    # 2 → no error, but some checks/remediation failed
+    if res.returncode not in (0, 2):
+        raise RuntimeError("oscap content evaluation failed")
+
+    # run openscap in chroot on the image tree (generate remediation script)
+    cmd = [
+        "/usr/sbin/chroot", tree,
+        "/usr/bin/oscap",
+        "xccdf", "generate"
+    ]
+
+    if verbose_level is not None:
+        cmd.extend(["--verbose", verbose_level])
+
+    if verbose_log is not None:
+        cmd.extend(["--verbose-log-file", f"{data_dir}/{verbose_log}.generate-fix"])
+
+    cmd.extend(["--profile", profile_id])
+    cmd.extend(["fix", "--fix-type", "bash",
+                "--output", f"{data_dir}/{REMEDIATION_SCRIPT}"])
+
+    cmd.append(f"{data_dir}/{xccdf_results}")
```

Comment:
@evgenyz going through this again, I have a qq about tailoring. Will this step need the tailoring file too if it's specified? Or are the generated results enough for this?

Reply:
Results should be enough (tailored-out rules are not processed and therefore won't be fixed).

Reply:
Apologies for re-opening this. I rebased some tailoring changes on this and notice that unfortunately this does break :/

With `composer` we overwrite the `profile_id` to the new tailoring `profile_id` (it just appends "_osbuild_tailoring" to the base profile). So this step breaks because it's looking for the tailoring profile.

I guess we could either: 
 - check for the suffix "_osbuild_tailoring" and save those into `base_profile_id` and `tailoring_profile_id`
 - add a `base_profile_id` in as a schema option, which is probably what I should have done in the first place.
 
The only problem I see with the second option is it breaks backwards compatibility. Maybe we could do both? Wdyt @thozza?

Reply:
@evgenyz is there any way we can use `oscap` to get the base `profile_id` from the tailoring file?

Reply:
So, with `oscap info tail.xml` you can get the base profile id. It's in a human-readable form, but should be stable.

Reply:
So there is no need to pass the tailoring profile id. I'll change my PR.

Reply:
Amazing, that will be the simplest. Thanks so much

Reply:
Well, actually it turned out that `generate fix` unlike other generators actually needs the same tailored file. Sorry for misguiding you. Nonetheless, I've updated the PR to reflect that. You should not change anything in the autotailor step, it does what it is supposed to do.